sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -259,6 +298,60 @@ public int hashCode() {
     }
   }
 
+  interface ColumnPosition {
+    First FIRST = new First();
+
+    static ColumnPosition After(String[] column) {
```

Comment:
`nameParts`? It'd be nice to add documentation that it refers to a nested field if String[] is longer than 1

Reply:
Also should AFTER take a multipart identifier? Should the syntax be:
```
ALTER TABLE ADD COLUMN m.n.y STRING AFTER x
```
where x is a column after the nested field `m.n.x`

Reply:
Can we have a different function name, @cloud-fan ?